src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -454,32 +463,80 @@ public override async Task<HttpRequestData> HandleRequestAsync(HttpStatusCode st
                 }
             }
 
-            List<string> headerLines = await AcceptConnectionSendResponseAndCloseAsync(statusCode, headerString, content);
-
+            List<string> headerLines = null;
             HttpRequestData requestData = new HttpRequestData();
 
-            // Parse method and path
-            string[] splits = headerLines[0].Split(' ');
-            requestData.Method = splits[0];
-            requestData.Path = splits[1];
-
-            // TODO: Add handling for request body.
-            // In the meantime, just fail any request that's not a GET so that we don't confuse
-            // the client by not consuming the request body.
-            if (requestData.Method != "GET")
+            await AcceptConnectionAsync(async connection =>
             {
-                throw new NotImplementedException("Request body not supported");
-            }
+                headerLines = await connection.ReadRequestHeaderAndSendResponseAsync(statusCode, headerString + "Connection: close\r\n", content);
```

Comment:
It seems weird to send the response before you read the request body. This may work ok for some tests, but in general it seems like we should do the normal thing here and read the body before sending the response.

Reply:
I agree. It was convenience to keep it as it was. I split the call and we send response now at the end.